fedimint-bin-tests/src/main.rs

```diff
@@ -0,0 +1,630 @@
+        let mut rpc = self.rpc.lock().await;
+        Ok(rpc.call_typed(request).await?)
+    }
+
+    pub async fn await_block_processing(&self) -> Result<()> {
+        loop {
+            let btc_height = self.bitcoind.client().get_blockchain_info()?.blocks;
+            let lnd_height = self
+                .request(cln_rpc::model::GetinfoRequest {})
+                .await?
+                .blockheight;
+            if (lnd_height as u64) == btc_height {
+                break;
+            }
+            info!("Lightningd::await_block_processing");
+            tokio::time::sleep(POLL_INTERVAL).await;
+        }
+        Ok(())
+    }
+
+    pub async fn pub_key(&self) -> Result<String> {
+        Ok(self
+            .request(cln_rpc::model::GetinfoRequest {})
+            .await?
+            .id
+            .to_string())
+    }
+}
+
+#[derive(Clone)]
+pub struct Lnd {
+    bitcoind: Bitcoind,
+    client: Arc<Mutex<tonic_lnd::LndClient>>,
+    process: ProcessHandle,
+}
+
+impl Lnd {
+    pub async fn new(process_mgr: &ProcessMgr, bitcoind: Bitcoind) -> Result<Self> {
+        let lnd_dir = env::var("FM_LND_DIR")?;
+        let cmd = cmd!("lnd", "--lnddir={lnd_dir}");
+
+        let process = process_mgr.spawn_daemon("lnd", cmd).await?;
+        let lnd_rpc_addr = env::var("FM_LND_RPC_ADDR")?;
+        let lnd_macaroon = env::var("FM_LND_MACAROON")?;
+        let lnd_tls_cert = env::var("FM_LND_TLS_CERT")?;
+        while !fs::try_exists(&lnd_tls_cert).await? {
+            info!("waiting for lnd start");
+            tokio::time::sleep(POLL_INTERVAL).await;
+        }
+        let client = tonic_lnd::connect(
+            lnd_rpc_addr.clone(),
+            lnd_tls_cert.clone(),
+            lnd_macaroon.clone(),
+        )
+        .await?;
+        Ok(Self {
+            bitcoind,
+            client: Arc::new(Mutex::new(client)),
+            process,
+        })
+    }
+
+    pub async fn client_lock(&self) -> Result<LndClient<'_>> {
+        let mut client = self.client.lock().await;
+        Ok(LndClient(client))
+    }
+
+    pub async fn pub_key(&self) -> Result<String> {
+        Ok(self
+            .client_lock()
+            .await?
+            .get_info(GetInfoRequest {})
+            .await?
+            .into_inner()
+            .identity_pubkey)
+    }
+
+    pub async fn await_block_processing(&self) -> Result<()> {
+        let mut client = self.client_lock().await?;
+        loop {
+            let info = client.get_info(GetInfoRequest {}).await?.into_inner();
+
+            if info.synced_to_chain {
+                break;
+            }
+
+            info!("Lnd::await_block_processing");
+            tokio::time::sleep(POLL_INTERVAL).await;
+        }
+        Ok(())
+    }
+}
+
+pub struct LndClient<'a>(MutexGuard<'a, tonic_lnd::LndClient>);
+
+impl Deref for LndClient<'_> {
+    type Target = tonic_lnd::LightningClient;
+
+    fn deref(&self) -> &Self::Target {
+        panic!("please don't deref mut");
```

Comment:
WAT. :D

This isn't `deref_mut`, and ... why have it... if you're not supposed to call it? (Also calling `deref` is super easy, so good luck to us).

Reply:
lets just remove it